server/tracker.go

```diff
@@ -66,42 +73,44 @@ func NewTracker(poolSize int, shufflePort int, shuffleWebSocketPort int, torShuf
 }
 
 // add adds a connection to the tracker.
-func (t *Tracker) add(data *trackerData) {
+func (t *Tracker) add(p *playerData) {
 	t.mutex.Lock()
 	defer t.mutex.Unlock()
 
-	t.verificationKeys[data.verificationKey] = data.conn
+	t.verificationKeys[p.verificationKey] = p.conn
 
-	data.sessionID = t.generateSessionID()
+	p.sessionID = t.generateSessionID()
 
-	t.connections[data.conn] = data
+	t.connections[p.conn] = p
 
-	data.pool, data.number = t.assignPool(data)
+	p.pool, p.number = t.assignPool(p)
 }
 
 // remove removes the connection.
 func (t *Tracker) remove(conn net.Conn) {
 	t.mutex.Lock()
 	defer t.mutex.Unlock()
 
-	td := t.connections[conn]
-	if td != nil {
-		if td.verificationKey != "" {
-			delete(t.verificationKeys, td.verificationKey)
+	player := t.connections[conn]
+	if player != nil {
+		if player.verificationKey != "" {
+			delete(t.verificationKeys, player.verificationKey)
 		}
 
-		t.unassignPool(td)
+		t.unassignPool(player)
 
 		delete(t.connections, conn)
 	}
 }
 
-// banned returns true if the player has been banned.
-func (t *Tracker) banned(data *trackerData) bool {
+// bannedByPool returns true if the player has been banned by their pool.
+func (t *Tracker) bannedByPool(p *playerData) bool {
 	t.mutex.RLock()
 	defer t.mutex.RUnlock()
 
-	return t.poolSizes[data.pool] <= (len(data.bannedBy) + 1)
+	blameCount := len(p.blamedBy)
```

Comment:
The old version read better to me. 

```
// Users can't blame themselves. =)
return t.poolVoters[data.pool] <= (len(data.blamedBy) + 1)
```

Reply:
I changed it because I had to go through several transformations before it clicked for me:

1. Reversal of the yoda condition (the relatively "fixed" value is on the left instead of the right)
2. Rather than removing a vote as suggested in the comment, it is adding one
3. Before "vote" was in the names, I actually had no idea what it was talking about until I started looking into the details of the variables.

Also my preference is to package conditionals so that they can be read in a single bite without additional mental parsing - letting the next coder understand at a glance or go back and check the details if they are worried about the implementation. Of course there are exceptions but as you mentioned, this is a critical condition so I thought it's important to make the intention as clear as possible.

Let me know what you prefer and I can update it.

Reply:
This does more allocations in memory than are needed. We can flip the comparison and use subtraction if you want. =)

Reply:
Now that the other changes are there, that does come close. With a little comment sprinkled on top, I think 3c19d80 does the job?

Reply:
Works for me!